test/date_range_test.rb

```diff
@@ -0,0 +1,76 @@
+require_relative "test_helper"
+
+describe "DateRange class" do
+  describe "DateRange instatiation" do
+    # before do
+    #   date_range = Hotel::DateRange.new()
+    # end
+    it "is an instance of DateRange" do
+      expect(Hotel::DateRange.new(start_date: "2020-3-1", end_date: "2020-3-10")).must_be_kind_of Hotel::DateRange
+    end
+
+    it "converts its start_date and end_date attributes to Date objects" do
+      date_range = Hotel::DateRange.new(start_date: "2020-3-1", end_date: "2020-3-10")
+      expect(date_range.start_date).must_be_kind_of Date
+      expect(date_range.end_date).must_be_kind_of Date
+    end 
+
+    it "raises an ArgumentError if end_date is equal to or comes before start_date" do
+      expect {Hotel::DateRange.new(start_date: "2020-3-1", end_date: "2020-2-10")}.must_raise ArgumentError
+      expect {Hotel::DateRange.new(start_date: "2020-3-1", end_date: "2020-3-1")}.must_raise ArgumentError
+    end
+
+    # TODO
+    # it "can be instantiated with no arguments" do
+    #   expect(Hotel::DateRange.new()).must_be_kind_of Hotel::DateRange
+    # end
+  end
+
+  describe "overlap?" do
+    let (:date_range) {
+      Hotel::DateRange.new(start_date: "2020-3-1", end_date: "2020-3-20")
+    }
+
+    it "expects a DateRange as an argument" do
+      expect {date_range.overlap?(50)}.must_raise ArgumentError
+    end
+
+    it "returns true if the dates overlap and false if not" do
+      # TODO: each one should be a new test?
```

Comment:
Since each other this `expect`s are testing a different way that dates could overlap, it would be good to put each of these assertions in their own `it` block that describes exactly what you are testing.